### ISSUE TITLE
Tensorflow failed to find 'TRTEngineOp' when building pip-package and libtensorflow_cc.so

### ISSUE THREAD

**Issue by qcraftai** [tensorflow/tensorflow#34238]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: 
- TensorFlow installed from (source or binary): source
- TensorFlow version: 1.14
- Python version: 3.6
- Installed using virtualenv? pip? conda?: Build from source
- Bazel version (if compiling from source): 0.24.1
- GCC/Compiler version (if compiling from source): gcc 7.4.0
- CUDA/cuDNN version:10.0
- GPU model and memory: 2080ti 11GB

**Describe the problem**

In brief, I built tensorflow for both python (pip_package) and c++ (libtensorflow_cc.so), I want to train a model, convert it to tf-trt in python, then run it on c++. And it failed when running it on c++.

Here are the details:

I'm trying to build tensorflow python version by bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package and build c++ version by bazel build --config=opt --config=monolithic --config=cuda //tensorflow:libtensorflow_cc.so. And I want them both with tensorrt support. After building it sequentially (configure first, build_pip_package and then libtensorflow_cc.so), I find that the python version could load tensorrt successfully, but when I'm trying to run session run on c++, it gives me the error saying:

```
Check failed: status.ok() Loading error: Op type not registered 'TRTEngineOp' in binary running on my_dev_docker. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you
are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
```

Since I'm using tf 1.14 where the tensorrt has been promoted to first-class citizen from tf.contrib. So I believe that I don't need to dynatically load it in my c++ code using c api "TF_LoadLibrary". So I started to suspect that maybe the tensorrt support is successfully installed but not on the c++ one and I didn't find any good reference on the internet on how to enable tf-trt for tensorflow_cc.so.

The success of python tensorflow with tensorrt support is tested by running "from tensorflow.python.compiler.tensorrt import trt_convert as trt". Although I have to run "export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/TensorRT-5.1.2.2/lib" first otherwise it will say "ImportError: libnvinfer.so.5: cannot open shared object file: No such file or directory".

The following is the exact command:

 PYTHON_BIN_PATH=/usr/bin/python3.6 \                                                                                                                                                                                                                                              
 PYTHON_LIB_PATH=/usr/local/lib/python3.6/dist-packages \                                                                                                                                                                                                                          
 TF_ENABLE_XLA=1 \                                                                                                                                                                                                                                                                 
 TF_NEED_OPENCL_SYCL=0 \                                                                                                                                                                                                                                                           
 TF_NEED_ROCM=0 \                                                                                                                                                                                                                                                                  
 TF_NEED_CUDA=1 \                                                                                                                                                                                                                                                                  
 TF_CUDA_VERSION=10 \                                                                                                                                                                                                                                                              
 TF_CUDA_PATHS=/usr/local/cuda,/usr/lib/x86_64-linux-gnu,/usr/include \                                                                                                                                                                                                            
 CUDA_TOOLKIT_PATH=/usr/local/cuda \                                                                                                                                                                                                                                               
 CUDNN_INSTALL_PATH=/usr/lib/x86_64-linux-gnu \                                                                                                                                                                                                                                    
 TF_CUDNN_VERSION=7 \                                                                                                                                                                                                                                                              
 TF_NEED_TENSORRT=1 \                                                                                                                                                                                                                                                              
 TF_TENSORRT_VERSION=5 \                                                                                                                                                                                                                                                           
 TENSORRT_INSTALL_PATH=/usr/local/TensorRT-5.1.2.2 \                                                                                                                                                                                                                               
 TF_NCCL_VERSION=2 \                                                                                                                                                                                                                                                               
 TF_CUDA_COMPUTE_CAPABILITIES="6.1,7.5" \                                                                                                                                                                                                                                          
 TF_CUDA_CLANG=0 \                                                                                                                                                                                                                                                                 
 TF_DOWNLOAD_CLANG=0 \                                                                                                                                                                                                                                                             
 GCC_HOST_COMPILER_PATH=/usr/bin/gcc \                                                                                                                                                                                                                                             
 CLANG_CUDA_COMPILER_PATH=/usr/local/clang_8.0.0/bin/clang \                                                                                                                                                                                                                       
 TF_NEED_MPI=0 \                                                                                                                                                                                                                                                                   
 CC_OPT_FLAGS="-mavx -Wno-sign-compare" \                                                                                                                                                                                                                                          
 TF_SET_ANDROID_WORKSPACE=0 \                                                                                                                                                                                                                                                      
 ./configure                                                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                                                   
 bazel build --config=opt --config=cuda //tensorflow/tools/pip_package:build_pip_package                                                                                                                                                                                           
 ./bazel-bin/tensorflow/tools/pip_package/build_pip_package /tmp/tensorflow_pkg                                                                                                                                                                                                    
 pip3 install /tmp/tensorflow_pkg/tensorflow-1.14.0-cp36-cp36m-linux_x86_64.whl                                                                                                                                                                                                    
                                                                                                                                                                                                                                                                                   
 bazel build --config=opt --config=monolithic --config=cuda //tensorflow:libtensorflow_cc.so

**Any other info / logs**

I run the conversion in python by:

from tensorflow.python.compiler.tensorrt import trt_convert as trt
converter = trt.TrtGraphConverter(input_saved_model_dir=input_saved_model_dir)
converter.convert()
converter.save(output_saved_model_dir)

I run the converted model in c++ by:

const auto status =                                                                                                                                                                                                                                                             
       LoadSavedModel(session_options, run_options, net_param_.model_path(),                                                                                                                                                                                                       
                      {tensorflow::kSavedModelTagServe}, &bundle_, gpu_id);

And I get error message from status.error_message();

Could you kindly provide some hints on what I might be doing wrong?

**Comment by ymodak** [comment]
Did you try importing ```tftrt``` module in addition to ```tensorflow``` module?
```python
 import the tftrt module in addition to import tensorflow
```

**Comment by aaroey** [comment]
Hi @qcraftai, I can see a dependency chain based on:
```
bazel query 'somepath(tensorflow:libtensorflow_cc.so, tensorflow/compiler/tf2tensorrt:trt_conversion)'
```

. When you run bazel build on tensorflow:libtensorflow_cc.so, can you try adding --config=tensorrt? Or make sure TensorRT is configured during the build by checking that environment variables like TF_NEED_TENSORRT and TENSORRT_INSTALL_PATH are set?

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
